includes/class-wc-stripe-intent-controller.php

```diff
@@ -891,26 +904,42 @@ public function is_mandate_data_required( $selected_payment_type ) {
 	/**
 	 * Creates and confirm a setup intent with the given payment method ID.
 	 *
-	 * @param string $payment_method The payment method ID (pm_).
+	 * @param array $payment_information The payment information to be used for the setup intent.
 	 *
 	 * @throws WC_Stripe_Exception If the create intent call returns with an error.
 	 *
 	 * @return array
 	 */
-	public function create_and_confirm_setup_intent( $payment_method ) {
-		// Determine the customer managing the payment methods, create one if we don't have one already.
-		$user        = wp_get_current_user();
-		$customer    = new WC_Stripe_Customer( $user->ID );
-		$customer_id = $customer->update_or_create_customer();
+	public function create_and_confirm_setup_intent( $payment_information ) {
+		$request  = [
+			'payment_method'       => $payment_information['payment_method'],
+			'payment_method_types' => [ $payment_information['selected_payment_type'] ],
+			'customer'             => $payment_information['customer'],
+			'confirm'              => 'true',
+		];
 
-		$setup_intent = WC_Stripe_API::request(
-			[
-				'customer'       => $customer_id,
-				'confirm'        => 'true',
-				'payment_method' => $payment_method,
-			],
-			'setup_intents'
-		);
+		// SEPA setup intents require mandate data.
+		if ( ! $this->is_mandate_data_required( $selected_payment_type ) ) {
+			$request = $this->add_mandate_data( $request );
+		}
+
+		/**
+		 * When the setup intent is being created from a checkout request, i.e. purchasing a subscription product
+		 * with a free trial, send the setup_intents request with level3 and order data.
+		 *
+		 * If the given $payment_information doesn't have the level3 or order data i.e. creating a setup intent from the Add Payment Method
+		 * page, send off a generic Stripe request.
+		 */
+		if ( isset( $payment_information['level3'], $payment_information['order'] ) ) {
```

Comment:
According to [Stripe's setup_intent documentation](https://stripe.com/docs/api/setup_intents/create), level3 data isn't a parameter of the create requrest, so I'll have to remove this.

Reply:
Removed in 6a589487da392b3c902cc60a0e9975fc23389809